doc/classes/Node.xml

```diff
@@ -578,235 +608,234 @@
+				Calls [method Object.notification] with [param what] on this node and all of its children, recursively.
 			</description>
 		</method>
 		<method name="queue_free">
 			<return type="void" />
 			<description>
-				Queues a node for deletion at the end of the current frame. When deleted, all of its child nodes will be deleted as well. This method ensures it's safe to delete the node, contrary to [method Object.free]. Use [method Object.is_queued_for_deletion] to check whether a node will be deleted at the end of the frame.
+				Queues this node to be deleted at the end of the current frame. When deleted, all of its children are deleted as well. Unlike [method Object.free], this method ensures it is safe to delete the node. Use [method Object.is_queued_for_deletion] to check if the node will be deleted at the end of the frame.
 			</description>
 		</method>
 		<method name="remove_child">
 			<return type="void" />
 			<param index="0" name="node" type="Node" />
 			<description>
-				Removes a child node. The node is NOT deleted and must be deleted manually.
-				[b]Note:[/b] This function may set the [member owner] of the removed Node (or its descendants) to be [code]null[/code], if that [member owner] is no longer a parent or ancestor.
+				Removes a child [param node]. The [param node], along with its children, are [b]not[/b] deleted. To delete a node, see [method queue_free].
+				[b]Note:[/b] This method sets the [member owner] of the removed [param node] (or its descendants) to [code]null[/code], if their [member owner] is no longer an ancestor (see [method is_ancestor_of]).
 			</description>
 		</method>
 		<method name="remove_from_group">
 			<return type="void" />
 			<param index="0" name="group" type="StringName" />
 			<description>
-				Removes a node from a group. See notes in the description, and the group methods in [SceneTree].
+				Removes the node from the given [param group]. Generates an error if the node has not been added to [param group]. To avoid this, use [method is_in_group] first. See also notes in the description, and the [SceneTree]'s group methods.
 			</description>
 		</method>
 		<method name="replace_by">
 			<return type="void" />
 			<param index="0" name="node" type="Node" />
 			<param index="1" name="keep_groups" type="bool" default="false" />
 			<description>
-				Replaces a node in a scene by the given one. Subscriptions that pass through this node will be lost.
-				If [param keep_groups] is [code]true[/code], the [param node] is added to the same groups that the replaced node is in.
-				[b]Note:[/b] The given node will become the new parent of any child nodes that the replaced node had.
-				[b]Note:[/b] The replaced node is not automatically freed, so you either need to keep it in a variable for later use or free it using [method Object.free].
+				Replaces this node by the given [param node]. All children of this node, as well as its signal connections, are moved to [param node].
```

Comment:
> Subscriptions that pass through this node will be lost.

I think this means if `foo.ready` signal was connected to `_on_foo_ready()`, after calling `foo.replace_by(x)`, `x.ready` is not connected to that function.

Reply:
Looking at the code again, what happens is, all connections of the original are lost, and the replacement is connected, instead. so when the signal is emitted, `original_node.my_method` is no longer called, but `replacement_node.my_method` is.

Reply:
With that said, I think this description makes sense? I'm a bit puzzled.

Reply:
What I mean is something like this:

```gdscript
var original_node := Node.new()
original_node.renamed.connect(_on_renamed)
assert(original_node.renamed.is_connected(_on_renamed))

var replacement_node := Node.new()
original_node.replace_by(replacement_node)

assert(original_node.renamed.is_connected(_on_renamed))  # still connected to original_node
assert(not replacement_node.renamed.is_connected(_on_renamed))  # connection considered lost
```

It's different from using "Change Type" in editor where connections are kept after the node is replaced by another.

Reply:
Take a look at the revision.